### PR TITLE
validate class and defined resource parameters

### PR DESCRIPTION
**Pull request by nod0n**
#### Pull Request (PR) description

This introduces two new type aliases: file_name and user_name

Validating parameters simplifies debugging and improves stability as
problematic parameters can be prohibited.

### CODE REVIEW

File: types/file_name.pp [hunk 1]
```diff
@@ -0,0 +1,2 @@
+# The name of a file. Not a full path!
```
Comment (bastelfreak) [inline types/file_name.pp:1]: please use `@summary` here as well so puppet-strings can pick it up

File: types/user_name.pp [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+#
+# From useradd(8): It is usually recommended to only use usernames
+# that begin with a lower case letter or an underscore, followed by lower case
+# letters, digits, underscores, or dashes. They can end with a dollar sign.
+# Usernames may only be up to 32 characters long.
+#
+# Many installations also allow capitals or periods, for example to separate
+# first and last names. 
+#
+type Sftp_jail::User_name =
```
Comment (bastelfreak) [inline types/user_name.pp:11]: please remove the linebreak after the `=`

File: types/file_name.pp (hunk 1 above)
Comment (nod0n) [reply types/file_name.pp]: fixed, REFERENCE.md updated

File: types/user_name.pp (hunk 2 above)
Comment (nod0n) [reply types/user_name.pp]: fixed

**Pull request merged**